### PR TITLE
Revert to instrumentation tests

### PR DESCRIPTION
**Pull request by yervantk**
### Changes
- Reverted back to Android Instrumentation tests. Tests still run as before.
- Retrofit network calls now happen on IO thread, and callbacks get returned on UI thread.

@krisorr @sav007

### CODE REVIEW

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java [hunk 1]
```diff
@@ -245,7 +247,7 @@ public void setPageSize(int pageSize) {
      * @param callback the {@link Callback} that will be used to indicate the response from the asynchronous network operation, not null
      */
     public void getShop(final Callback<Shop> callback) {
-        retrofitService.getShop().subscribe(new InternalCallback<Shop>() {
+        retrofitService.getShop().observeOn(AndroidSchedulers.mainThread()).subscribe(new InternalCallback<Shop>() {
```
Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java]: Lets extract AndroidSchedulers.mainThread() as local field so in future we can configure what thread/scheduler client want to use as for call back.

**Comment by krisorr** [comment]
@sav007, @yervantk - lets get the branch back in working order, then look at updating with annotations in a seperate PR

File: MobileBuy/buy/build.gradle [hunk 2]
```diff
@@ -128,12 +128,6 @@ dependencies {
     androidTestCompile 'com.google.dexmaker:dexmaker:1.1'
     androidTestCompile 'com.google.dexmaker:dexmaker-mockito:1.1'
 
-
-    testCompile 'org.mockito:mockito-core:1.9.5'
```
Comment (krisorr) [inline MobileBuy/buy/build.gradle]: I think the testUsingMockito should fail if you are missing this import

Comment (yervantk) [inline MobileBuy/buy/build.gradle]: This is "testCompile" instead of "androidTestCompile", so it's not needed anymore since we got rid of JUnit tests.

**Comment by krisorr** [comment]
LGTM

**Pull request merged**